imagery/i.atcorr/create_iwave.py

```diff
@@ -62,20 +59,15 @@ def read_input(csvfile):
     first column is wavelength
     values are those of the discrete band filter functions
     """
-    infile = open(csvfile, "r")
+    with open(csvfile, "r") as infile:
+        # get number of bands and band names
+        bands = infile.readline().strip().split(",")[1:]
 
-    # get number of bands and band names
-    bands = infile.readline().split(",")
-    bands.remove(bands[0])
-    bands[-1] = bands[-1].strip()
-    print(" > Number of bands found: %d" % len(bands))
-    infile.close()
+    print(f" > Number of bands found: {len(bands)}")
```

Comment:
A linter should complain that bands could be undefined here.

Reply:
Are you sure that `bands` could be undefined here? Aren't they always defined two lines before? If `csvfile` does not exist, the line would simply fail.

Reply:
I don't know why I was saying that, I know now for sure that `with` in Python doesn't create a scope for the variables, the variables defined inside are available after leaving the block. An attempt to reuse the closed ressource would fail thought, but it isn't the case here.